brion/io/morphologies.cpp

```diff
@@ -0,0 +1,136 @@
+/* Copyright (c) 2013-2015, EPFL/Blue Brain Project
+ *                          Juan Hernando <jhernando@fi.upm.es>
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "morphologies.h"
+#include "morphology.h"
+
+#include "brion/circuit.h"
+#include "brion/blueConfig.h"
+
+#include <lunchbox/log.h>
+
+namespace brion
+{
+namespace io
+{
+namespace morphologies
+{
+
+typedef std::map<std::string, boost::shared_ptr< Morphology > > MorphologyMap;
+
+Morphologies load( const BlueConfig& config, const GIDSet& gids )
+{
+    const brion::Circuit circuit( config.getCircuitSource( ));
+    return load( circuit, config.getMorphologySource(), gids );
+}
+
+Morphologies load( const BlueConfig& config, GIDSet& gids )
+{
+    brion::Circuit circuit( config.getCircuitSource( ));
+    return load( circuit, config.getMorphologySource(), gids );
+}
```

Comment:
We've had so many times the discussion about throw versus warn that I decided to implement the two flavours. The non-const version being preferred is consistent with not throwing by default, so unless you've changed your mind about that I'll keep it like this. It also puts the burden of verifying that the set sizes are the same on the client, but that's the price you have to pay for not throwing. It's a matter of trade-offs and I don't know if there's a one size fits all.